### PR TITLE
add description to timeout messages

### PR DESCRIPTION
**Pull request by marekbiskup**

### CODE REVIEW

**Comment by marekbiskup** [comment]
@quinton-hoole , @lavalamp you reviewed my previous attempt. Maybe you could also have a look at this one. Thanks

File: test/e2e/util.go [hunk 1]
```diff
@@ -517,32 +517,53 @@ func waitForRCPodToDisappear(c *client.Client, ns, rcName, podName string) error
 	})
 }
 
-// waits until the service appears (exists == true), or disappears (exists == false)
-func waitForService(c *client.Client, namespace, name string, exist bool, interval, timeout time.Duration) error {
-	return wait.Poll(interval, timeout, func() (bool, error) {
+// waitForService waits until the service appears (toAppear == true), or disappears (toAppear == false)
+func waitForService(c *client.Client, namespace, name string, toAppear bool, interval, timeout time.Duration) error {
```
Comment (lavalamp) [inline test/e2e/util.go]: IMO exist is clearer than toAppear. but this is a pretty minor nit.

File: test/e2e/util.go [hunk 2]
```diff
@@ -517,32 +517,53 @@ func waitForRCPodToDisappear(c *client.Client, ns, rcName, podName string) error
...
-			return !exist, nil
+			return !toAppear, nil
 		} else {
 			Logf("Service %s in namespace %s found.", name, namespace)
-			return exist, nil
+			return toAppear, nil
 		}
 	})
+	if err != nil {
+		var desiredStatus string
```
Comment (lavalamp) [inline test/e2e/util.go]: I have seen the pattern ```map[bool]{true: "to appear", false: "to disappear"}[toAppear]```. Not saying it's better, just tossing it out as an option.

File: test/e2e/util.go [hunk 3]
```diff
@@ -517,32 +517,53 @@ func waitForRCPodToDisappear(c *client.Client, ns, rcName, podName string) error
...
+		if toAppear {
+			desiredStatus = " to appear"
+		} else {
+			desiredStatus = " to dissappear"
```
Comment (lavalamp) [inline test/e2e/util.go]: tiny tiny nit: if you change anything else, can you take the space out of this and put it in the Errorf below? I think it'd be a tiny bit easier to read...

**Comment by lavalamp** [comment]
LGTM, thanks

**Comment by quinton-hoole** [comment]
LGTM.  @marekbiskup let me know when you've addressed the nits and we can mark as ok-to-merge for the oncall.

File: test/e2e/util.go (hunk 1 above)
Comment (marekbiskup) [inline test/e2e/util.go]: I'll change it back

File: test/e2e/util.go (hunk 2 above)
Comment (marekbiskup) [inline test/e2e/util.go]: Indeed, much better.

File: test/e2e/util.go (hunk 3 above)
Comment (marekbiskup) [inline test/e2e/util.go]: done

**Comment by marekbiskup** [comment]
@quinton-hoole , I addressed the nits. Test passed. Please check the patch and merge.

**Comment by quinton-hoole** [comment]
LGTM!  Thanks @marekbiskup

**Pull request merged**